### PR TITLE
#164798289 Email Verification

### PR DESCRIPTION
**Pull request by patrickf949**
#### What does this PR do?
Ensures a user receives an activation email upon registration to verify the ownership of the email.

#### Description of Task to be completed?
- Write tests for email verification
- Add Authors haven's email details to settings file
- Add email message and recipients to registration route
- Add activation link in email

#### How should this be manually tested?

After cloning the repo, cd into it and run the following commands to activate the virtual environment
- `$ virtualenv venv`
- `$ source venv/bin/activate`

Install all requirements
- `$ pip install -r requirements.txt`

Make migrations
- `$ python manage.py makemigrations`
- `$ python manage.py migrate`

Run the server 
- `$ python manage.py runserver`

Open Postman and Register a user using the following format and providing a valid email address:
Add a request using this route

- `localhost:8000/api/users/register/`

Set the method to POST
Click on body, Select raw and choose JSON(application/json) as your data format
Pass your details in this format providing a valid gmail account
```
{
    "user": {
                  "email": "your email address",
                  "password": "your valid password",
                  "username": "your preferred username"
       }
}
```
Check your email and use the link provided to activate your account

After this you should be able to login to your account

#### What are the relevant pivotal tracker stories?
[#164798289](https://www.pivotaltracker.com/story/show/164798289)

#### Screenshots

- Postman

Screenshot 2019-04-03 at 10 47 35

- Email
Screenshot 2019-04-04 at 22 29 53

### CODE REVIEW

**Comment by inno-asiimwe** [comment]
@patrickf949 the title to the `PR` description does not properly represent our intentions.

**Review by Genza999 — APPROVED** [review]
Great work Patrick. Good you removed all the commented out code.

**Review by e-ian — APPROVED** [review]
LGTM

File: authors/apps/authentication/models.py [hunk 1]
```diff
@@ -33,6 +36,16 @@ def create_user(self, username, email, password=None):
         user.set_password(password)
         user.save()
 
+        
+        
+        # send_mail(
```
Comment (inno-asiimwe) [inline authors/apps/authentication/models.py]: Remove the commented out code here.

Comment (patrickf949) [reply authors/apps/authentication/models.py]: @inno-asiimwe Thanks, I've implemented this

**Pull request merged**